examples/pytorch_mnist.py

```diff
@@ -294,3 +294,4 @@ def test(record):
     train(epoch)
 test(test_record)
 print(f"[{bf.rank()}]: ", test_record)
+bf.barrier()
```

Comment:
Do we need a barrier before test(test_record)?

Reply:
In case, one process finished too early and bring down the whole MPI processes.